File: phase-2/backend/schemas/task.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class TaskBase(BaseModel):
    title: str
    description: Optional[str] = None
    completion_status: bool = False


class TaskCreate(TaskBase):
    title: str  # Required field


class TaskUpdate(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    completion_status: Optional[bool] = None
    version: Optional[int] = None  # For optimistic locking


class TaskRead(TaskBase):
    id: str
    user_id: str
    created_at: datetime
    updated_at: datetime
    version: int

    class Config:
        from_attributes = True

File: phase-3/frontend/src/types/task.ts
export interface Task {
  id: string;
  title: string;
  description?: string;
  completion_status: boolean;
  user_id: string;
  created_at: string;
  updated_at: string;
}

export interface UserPreference {
  id: string;
  user_id: string;
  preference_key: string;
  preference_value: string;
  created_at: string;
  updated_at: string;
}